src/BridgeGuest.ts

```diff
@@ -0,0 +1,85 @@
+import { SourceBridgeClient } from './SourceBridgeClient'
+import { generateRequestId } from './generateRequestId'
+import { AnyEventHandler, Auth, AuthenticationResponse, Event, HelloResponse } from './types'
+
+interface BridgeGuestOptions {
+  eventHandlers: Record<string, AnyEventHandler>
+  autoReady: boolean
+}
+
+class BridgeGuestAPI {
+  private client: SourceBridgeClient
+
+  constructor(private readonly otherWindow: Window) {
+    this.client = new SourceBridgeClient(otherWindow)
+  }
+
+  // TODO should this return anything, really? return the hello response?
```

Comment:
no I think.

Reply:
I actually think it should return the hello response, since that is useful for building application protocols, but I won't make that change yet, we can do it if we need it for the connect apps.